config/v1/types_ingress.go

```diff
@@ -41,13 +43,128 @@ type IngressSpec struct {
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// hostname is the hostname that should be used by the route.
+	// +kubebuilder:validation:Required
+	// +required
+	Hostname hostname `json:"hostname"`
+
+	// servingCertKeyPairSecret is a reference to a secret of type `kubernetes.io/tls` in the openshift-config namespace.
+	// The serving cert/key pair must match and will be used by the operator to fulfill the intent of serving with this name.
+	// +optional
+	ServingCertKeyPairSecret SecretNameReference `json:"servingCertKeyPairSecret"`
+}
+
+// ComponentRouteStatus contains information allowing configuration of a route's hostname and serving certificate.
+type ComponentRouteStatus struct {
+	// namespace is the namespace of the route to customize. It must be a real namespace. Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	//
+	// The namespace and name of this componentRoute must match a corresponding
+	// entry in the list of spec.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
+
+	// name is the logical name of the route to customize. It does not have to be the actual name of a route resource
+	// but it cannot be renamed.
+	//
+	// The namespace and name of this componentRoute must match a corresponding
+	// entry in the list of spec.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// defaultHostname is the hostname of this route prior to customization.
+	// +kubebuilder:validation:Required
+	// +required
+	DefaultHostname hostname `json:"defaultHostname"`
+
+	// consumingUsers is a slice of ServiceAccounts that need to have read permission on the servingCertKeyPairSecret secret.
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	ConsumingUsers []consumingUser `json:"consumingUsers,omitempty"`
 
+	// currentHostnames is the list of current names used by the route.
+	// +kubebuilder:validation:MinItems=1
+	// +optional
+	CurrentHostnames []hostname `json:"currentHostnames,omitempty"`
+
+	// conditions are used to communicate the state of the componentRoutes entry.
+	//
+	// Supported conditions include Degraded and Progressing.
+	// If Degraded is true, that means something has gone wrong trying to handle the componentRoutes entry.
+	// The currentHostnames field may or may not be in effect.
+	//
+	// If Progressing is true, that means the component is taking some action related to the componentRoutes entry.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Is there a condition that states that the componentRoute is OK ? eg. `ComponentRouteAvailable` ?

Reply:
I see why communicating the availability of the default route while progressing to the configured route makes sense in your [existing usecase](https://github.com/openshift/api/pull/852#pullrequestreview-607391725). If we decide to follow your approach and redirect traffic from the existing route that may make sense, however, if configuring a route nukes the default route it may not as the available condition can be inferred from the degraded/progressing conditions.

Reply:
I've added documentation on the Available condition given the usecase you described.